src/storage/store/chunked_node_group.cpp

```diff
@@ -13,16 +13,25 @@ namespace kuzu {
 namespace storage {
 
 ChunkedNodeGroup::ChunkedNodeGroup(std::vector<std::unique_ptr<ColumnChunk>> chunks,
-    row_idx_t startRowIdx, NodeGroupDataFormat format)
+    row_idx_t startRowIdx, NodeGroupDataFormat format, bool hasExtraData)
     : format{format}, startRowIdx{startRowIdx}, chunks{std::move(chunks)} {
     KU_ASSERT(!this->chunks.empty());
+
     residencyState = this->chunks[0]->getResidencyState();
     numRows = this->chunks[0]->getNumValues();
     capacity = numRows;
     for (auto columnID = 1u; columnID < this->chunks.size(); columnID++) {
         KU_ASSERT(this->chunks[columnID]->getNumValues() == numRows);
         KU_ASSERT(this->chunks[columnID]->getResidencyState() == residencyState);
     }
+
+    if (hasExtraData) {
+        extraData = decltype(extraData)::value_type{};
+        for (auto columnType : CopyConstants::WARNING_METADATA_COLUMN_TYPES) {
+            extraData->emplace_back(std::make_unique<ColumnChunk>(LogicalType(columnType), capacity,
+                this->chunks[0]->isCompressionEnabled(), residencyState));
+        }
+    }
```

Comment:
It looks like this constructor is only called during `flushAsNewChunkedNodeGroup` and `deserialize`. In this case would it make sense to leave `extraData` empty since I don't think we need it in these use cases?

Reply:
Removing this for now since we don't actually have any code paths that make use of the `extraData` in this constructor, let me know if it is something we should leave in